packages/permission-controller/src/PermissionController.ts

```diff
@@ -1636,7 +1762,15 @@ export class PermissionController<
           performCaveatValidation: false,
         });
       }
-      permissions[targetName] = permission;
+
+      if (mergePermissions) {
+        permissions[targetName] = this.#mergePermission(
```

Comment:
In `requestPermissionsIncremental` this requires a validation step afterwards, do we need one here as well?

Reply:
We absolutely do, great catch: [38885ac](https://github.com/MetaMask/core/pull/4222/commits/38885acbc9a4a57facc72dfcfeeae34b2d570df4)